rest/src/main/kotlin/systems/ajax/englishstudytelegrambot/service/WordService.kt

```diff
@@ -25,37 +52,118 @@ class WordServiceImpl(
     val additionalInfoAboutWordService: AdditionalInfoAboutWordService
 ) : WordService {
 
-    override suspend fun saveNewWord(libraryName: String, telegramUserId: String, wordDto: WordDto): Word {
-        val libraryId = libraryRepository.getLibraryIdByLibraryNameAndTelegramUserId(libraryName, telegramUserId)
-        if (wordRepository.isWordBelongsToLibraryByWordSpelling(wordDto.spelling, libraryId))
-            throw WordAlreadyPresentInLibraryException()
-
-        val additionalInfo = additionalInfoAboutWordService.findAdditionInfoAboutWord(wordDto.spelling)
-        return wordRepository.saveNewWord(
-            Word(
-                spelling = wordDto.spelling,
-                translate = wordDto.translate,
-                libraryId = libraryId,
-                additionalInfoAboutWord = additionalInfo
+    override fun saveNewWord(
+        libraryName: String,
+        telegramUserId: String,
+        createWordDtoRequest: CreateWordDtoRequest
+    ): Mono<WordDtoResponse> =
+        Mono
+            .zip(
+                findLibraryIdWithoutCertainSpelling(libraryName, telegramUserId, createWordDtoRequest.spelling),
+                additionalInfoAboutWordService.findAdditionInfoAboutWord(createWordDtoRequest.spelling)
             )
-        )
-    }
+            .flatMap { (libraryId, additionalInfoAboutWord) ->
+                collectAndSaveWordInDb(createWordDtoRequest, libraryId, additionalInfoAboutWord)
+            }
+            .map(Word::toDtoResponse)
 
-    override fun updateWordTranslate(libraryName: String, telegramUserId: String, wordDto: WordDto): Word {
-        val wordId = wordRepository.getWordIdByLibraryNameTelegramUserIdWordSpelling(
-            libraryName,
-            telegramUserId,
-            wordDto.spelling
-        )
-        return wordRepository.updateWordTranslating(wordId, wordDto.translate)
-    }
+    override fun updateWordTranslate(
+        libraryName: String,
+        telegramUserId: String,
+        createWordDtoRequest: CreateWordDtoRequest
+    ): Mono<WordDtoResponse> =
+        wordRepository
+            .getWordIdByLibraryNameTelegramUserIdWordSpelling(
+                libraryName,
+                telegramUserId,
+                createWordDtoRequest.spelling
+            )
+            .switchIfEmpty(
+                checkWordOrLibraryIsMissing(libraryName, telegramUserId)
+            )
+            .flatMap { wordId ->
+                wordRepository.updateWordTranslating(wordId, createWordDtoRequest.translate)
+            }
```

Comment:
```suggestion
            .switchIfEmpty(checkWordOrLibraryIsMissing(libraryName, telegramUserId))
            .flatMap { wordRepository.updateWordTranslating(it, createWordDtoRequest.translate) }
```
[nit] I prefer having reactor operator on one line if possible, it's easier to read and understand what's happening